src/plugins/ui_feature/insert_pivot.ts

```diff
@@ -4,12 +4,26 @@ import { SpreadsheetPivotTable } from "../../helpers/pivot/table_spreadsheet_piv
 import { getZoneArea, positionToZone } from "../../helpers/zones";
 import { _t } from "../../translation";
 import { CellPosition, HeaderIndex, PivotTableCell, PivotTableData, UID, Zone } from "../../types";
-import { Command } from "../../types/commands";
+import { Command, CommandResult } from "../../types/commands";
 import { UIPlugin } from "../ui_plugin";
 
 export class InsertPivotPlugin extends UIPlugin {
   static getters = [] as const;
 
+  allowDispatch(cmd: Command) {
+    switch (cmd.type) {
+      case "DUPLICATE_PIVOT_IN_NEW_SHEET":
+        if (!this.getters.isExistingPivot(cmd.pivotId)) {
```

Comment:
I might be restarting an old discussion (sorry if I do) but the "insert_new_pivot" command pretty much does the same thing but does not crash because the 'getTableStructure' won't crash before the pivot is evaluated.
Couldn't "getTableStructure" not throw at all?